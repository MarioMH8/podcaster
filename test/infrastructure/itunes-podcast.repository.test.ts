import { SearchPodcastCriteria } from '@domain';
import { ItunesPodcastParser, ITunesPodcastRepository } from '@infrastructure';
import server from '@mock/msw/node';
import { afterAll, afterEach, beforeAll, beforeEach, describe, expect, it } from 'vitest';

describe('ItunesPodcastRepository', () => {
	let repository: ITunesPodcastRepository;
	let parser: ItunesPodcastParser;
	beforeAll(() => {
		server.listen();
	});
	beforeEach(() => {
		parser = new ItunesPodcastParser();
		repository = new ITunesPodcastRepository(parser);
	});
	afterEach(() => {
		server.dispose();
	});
	afterAll(() => {
		server.close();
	});

	it('should fetch all podcast successfully', async () => {
		const criteria = new SearchPodcastCriteria({ genre: 1, limit: 10 });

		const podcast = await repository.search(criteria);

		expect(podcast.map(p => p.toPrimitives())).toMatchSnapshot();
	});
});
