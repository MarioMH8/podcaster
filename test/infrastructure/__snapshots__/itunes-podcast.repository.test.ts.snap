// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ItunesPodcastRepository > should fetch all podcast successfully 1`] = `
[
  {
    "author": "Tony Mantor",
    "description": "Nashville based veteran music producer Tony Mantor’s Almost Live… Nashville podcast takes its listener on a journey through the business of music and entertainment. Following the success and popularity of his top-rated podcast on Autism & Mental Health awareness - Why Not Me? - Mantor hosts thought-provoking conversations and behind-the-scenes stories from various celebrity guests from the world of music, television and entertainment. With more than 30 years as an accomplished music producer based in Nashville, Mantor welcomes a worldly guest list of musicians, actors and entertainers who share their unique stories about the roles they’ve played in creating and sustaining what we know as popular American culture. Conversations and stories that also give a deeper understanding on what it takes to achieve success in the entertainment industry for aspiring entertainers who are currently pursuing their own dreams of being a star someday. Whether listening for the entertainment or storytelling factor alone, or for tips on how his guests faced their life’s challenges, Almost Live… Nashville will have a story for anyone who has been a fan of popular culture in music, movies and television over the last few decades.",
    "id": 1751194045,
    "name": "'Tony Mantor's : Almost Live..... Nashville'",
  },
  {
    "author": "The Joe Budden Network",
    "description": "Tune into Joe Budden and his friends. Follow along the crazy adventures of these very random friends.",
    "id": 1535809341,
    "name": "The Joe Budden Podcast",
  },
]
`;
